### PR TITLE
Blank map on event projection switch fix

### LINKED ISSUE #2865: When switching from events in Geographic to Antarctic, the map goes blank
**Describe the bug**
When switching from an event that appears both in geographic to antarctic, e.g. Iceberg B49, the map goes blank when you switch to Antarctic. Seems like it happens on the initial click to Antarctic, if you click on Antarctic again, the map populates.

**To Reproduce**
1. In Geographic, Click on Events, go down the list and click on Iceberg B49.
2. Change to Antarctic.
3. See error - map is blank. Console error says “TypeError: pixel2 is null” in Firefox and “TypeError: Cannot read property ‘0’ of null” in Chrome. Seems like it happens on the initial click to Antarctic, if you click on Antarctic again, the map populates.

**Expected behavior**
The map in Antarctic should not go blank. 

**Desktop (please complete the following information):**
 - OS: OSX
 - Browser: Chrome, Firefox

### PR DESCRIPTION
**Pull request by jasontk19**
## Description

Fixes #2865 

- Some code cleanup
- Prevent switching to projection that is already active (was causing issue re-rendering event tracks
- Fix issue with map going blank on projection switch while event is selected
- [ ] update test suite

Bonus:
- Add zoomToEvent when proj switches with event selected
- Update deprecated stylelint rule names

## How to test

1. Select "Events" tab
2. Select a sea ice event near arctic or antarctic with multiple dates
3. Select a date other than the default
4. Switch to appropriate polar projection
5. Confirm map shows
6. Confirm map zooms to the selected event
7. Confirm the same date is selected in the sidebar and the corresponding point is centered on the map

### CODE REVIEW

**Review by minniewong — COMMENTED** [review]
Looking at the test plan (events.focus-1 thru 5), if you select an event in Geographic and switch projections, I don't think that it's supposed to zoom to the event upon changing projections.

**Comment by jasontk19** [comment]
> Looking at the test plan (events.focus-1 thru 5), if you select an event in Geographic and switch projections, I don't think that it's supposed to zoom to the event upon changing projections.

Hmmm... any idea why not?  Seemed like a nice thing to have 🤷

**Comment by minniewong** [comment]
Not sure why! perhaps we can discuss at the meeting, and figure out why :)

**Comment by jasontk19** [comment]
Added it to the agenda.  I'm guessing/hoping the test suite is just worded that way to accurately reflect how the app was working previously. 🤞

File: web/js/map/natural-events/track.js [hunk 1]
```diff
@@ -148,26 +149,12 @@ export default function naturalEventsTrack(ui, store, selectedMap) {
...
       // but there is a multiday event, build a new track
-      newTrackDetails = createTrack(
-        proj,
-        event,
-        selectedMap,
-        selectedDate,
-        callback,
-      );
-      selectedMap.addLayer(newTrackDetails.track);
+      createAndAddTrack();
```
Comment (Benjaki2) [inline web/js/map/natural-events/track.js:157]: 👍

**Review by Benjaki2 — APPROVED** [review]
Looks good

**Review by minniewong — APPROVED** [review]
Looking good! 😎

**Pull request merged**